otelx/otel_test.go

```diff
@@ -0,0 +1,53 @@
+package otelx
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"testing"
+
+	"github.com/ory/x/logrusx"
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/otel/attribute"
+)
+
+func TestJaegerTracer(t *testing.T) {
+	host := "127.0.0.1"
+	port := "6831"
+	done := make(chan struct{})
+	go func(addr string) {
+		t.Log("Starting test UDP server for Jaeger spans")
+		udpAddr, _ := net.ResolveUDPAddr("udp", addr)
+		srv, _ := net.ListenUDP("udp", udpAddr)
+		for {
+			buf := make([]byte, 2048)
+			_, conn, _ := srv.ReadFromUDP(buf)
+			if conn == nil {
+				continue
+			}
+			if len(buf) != 0 {
+				t.Log("recieved span!")
+				done <- struct{}{}
+			}
+			break
+		}
+	}(fmt.Sprintf("%s:%s", host, port))
+
+	ot, err := New("github.com/ory/x/otelx", logrusx.New("ory/x", "1"), &Config{
+		ServiceName: "Ory X",
+		Provider:    "jaeger",
+		Providers: ProvidersConfig{
+			Jaeger: JaegerConfig{
+				LocalAgentAddress: "127.0.0.1:6831",
+			},
+		},
+	})
+	require.NoError(t, err)
+
+	trc := ot.Tracer()
+	_, span := trc.Start(context.Background(), "testSpan")
+	span.SetAttributes(attribute.Bool("testAttribute", true))
+	span.End()
+
+	<-done
```

Comment:
This means, in case of failure, the test just runs into timeout right? That's not very nice for debugging/CI.
Easy fix:
```suggestion
	select {
		case <-done:
		case <-time.After(10*time.Second):
			t.Fail("expected to receive span, but did not receive any")
	}
```

Reply:
This was how I'd written it initially, but sometimes, it takes a while for the span to come through (for whatever reason) and it's hard to accurately predict the duration. I've gone ahead and added a 15s limit, but this could potentially introduce flakiness.